rfcs/0007-additional-type-representations.md

```diff
@@ -0,0 +1,58 @@
+# Additional Type Representations
+
+## Purpose
+
+The current list of scalar type representations is inadequate:
+
+- JSON integer and number types are not what we want for most applications, since they are represented by an unusual 53-bit number type. Since scalars can be both inputs and outputs, and are therefore invariant, we cannot safely convert to and from larger and smaller integer and floating point types.
+- We want to be able to generate some standard scalar types in Hasura for types which are used across different data sources, e.g. UUIDs. This needs support in NDC so that we can generate the right types on the Hasura side (or perform accurate code generation in general).
+- We want to be able to support additional transport mechanisms later (e.g. GRPC) and we will want to guarantee that responses do not depend on the transport mechanism used. It is not enough to simply say a type has no concrete representation, because even if data is "arbitrary JSON", we need to be able to encode that as GRPC later. Therefore, we want to make representation required, but call out two new abstract representations: JSON, and bytes.
+
+## Proposal
+
+### New representations
+
+Add the following type representations to `ndc_models::TypeRepresentation`:
+
+|Name|Description|JSON representation|
+|-|-|-|
+| Int8 | A 8-bit signed integer with a minimum value of -2^7 and a maximum value of 2^7 - 1 | Number |
+| Int16 | A 16-bit signed integer with a minimum value of -2^15 and a maximum value of 2^15 - 1 | Number |
+| Int32 | A 32-bit signed integer with a minimum value of -2^31 and a maximum value of 2^31 - 1 | Number |
+| Int64 | A 64-bit signed integer with a minimum value of -2^63 and a maximum value of 2^63 - 1 | String |
+| Float32 | An IEEE-754 single-precision floating-point number | Number |
+| Float64 | An IEEE-754 double-precision floating-point number | String |
```

Comment:
Shouldn't Float64 be Number instead of String?
In v2, float64 postgres columns are returned as JSON numbers.
Javascript number is also a Float64 so I think all libraries should be able to interpret number as a Float64.